keyboards/pimoroni/keybow2040/readme.md

```diff
@@ -0,0 +1,83 @@
+Keybow 2040
+===========
+
+![Keybow 2040](https://cdn.learn.pimoroni.com/article/assembling-keybow-2040/assets/keybow2040-03.jpg?width=800)
+
+_This QMK configuration was created by Tony Finch <dot@dotat.at>, **not** by Pimoroni._
+
+The [Pimoroni Keybow 2040][keybow2040] is a 16 key mechanical macropad
+controlled by a Raspberry Pi RP2040, with Kailh hot swap sockets and
+per key RGB lighting, in a decorative FR4 sandwich.
+
+The Keybow 2040 is also known by its codename Venus Dust, or its
+product codes of the form PIM56x.
+
+[keybow2040]: https://shop.pimoroni.com/products/keybow-2040
+
+Make example for this keyboard (after setting up your build environment):
+
+    make pimoroni/keybow2040:default
+
+Flashing example for this keyboard:
+
+    make pimoroni/keybow2040:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools)
+and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide)
+for more information. Brand new to QMK? Start with our
+[Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+
+notes
+-----
+
+  * The BOOTSEL button doubles as a 17th configurable key.
+
+  * There is no USB PID allocated for the Keybow 2040 under
+    [the Raspberry Pi USB VID](https://github.com/raspberrypi/usb-pid),
+    but its CircuitPython port uses VID = 0x16D0, PID = 0x08C6
+    so I copied that.
```

Comment:
I'm not sure whether copying the USB ID from CircuitPython is the proper thing to do — this may create some conflicts, like VIA trying to poke a board that uses the CircuitPython firmware and therefore doing some unexpected things.

Reply:
Oh good, I thought it might be a bad idea too. (Also the CircuitPython VID is a real VID and the PID only fails to collide by luck.) I will go back to my original plan of cooking up a new “random” pair of IDs.

Reply:
0x16D0 is the VID for which unique PID values are sold by [MCS Electronics](https://www.mcselec.com/index.php?page=shop.product_details&product_id=92&option=com_phpshop), so the PID can probably be considered as “owned” by Pimoroni.

The problem with VIA is that it does not perform a proper detection for the HID usage page to find the rawhid interface (although that usage page is in the vendor range, so in theory it might collide with some completely unrelated usage).  Another potential problem is that Windows in some circumstances can cache the binding of drivers to USB interfaces, therefore you may have some trouble if the device changes the set of advertised interfaces without changing its USB VID:PID pair.

Reply:
Oh, I didn’t know MCS did that, cool. And thanks for those details about why it is best to avoid this kind of collision.

Reply:
I picked new USB IDs in round two, and I checked [it does not collide](https://devicehunt.com/search/type/pci/vendor/07F8/device/any)